app/components/Views/Settings/Contacts/ContactForm/index.js

```diff
@@ -178,14 +179,19 @@ class ContactForm extends PureComponent {
         this.setState({ inputWidth: '100%' });
       }, 100);
     if (mode === EDIT) {
-      const { addressBook, chainId, identities } = this.props;
+      const { addressBook, chainId, internalAccounts } = this.props;
       const networkAddressBook = addressBook[chainId] || {};
       const address = this.props.route.params?.address ?? '';
-      const contact = networkAddressBook[address] || identities[address];
+      const contact =
+        networkAddressBook[address] ||
+        (address &&
+          internalAccounts.find((account) =>
+            toLowerCaseEquals(account.address, address),
+          ));
       this.setState({
         address,
-        name: contact.name,
-        memo: contact.memo,
+        name: contact?.name || '',
+        memo: contact?.memo || '',
```

Comment:
```suggestion
        name: contact?.name ?? '',
        memo: contact?.memo ?? '',
```

not sure if it makes sense but this way we allow the name and memo to be 0 and false